E0302 needs to be updated to new format

From: src/test/compile-fail/E0302.rs

E0302 needs a span_label, updating it from:

```
error[E0302]: cannot assign in a pattern guard
  --> src/test/compile-fail/E0302.rs:14:21
   |
14 |         option if { option = None; false } => { }, //~ ERROR E0302
   |                     ^^^^^^^^^^^^^
```

To:

```
error[E0302]: cannot assign in a pattern guard
  --> src/test/compile-fail/E0302.rs:14:21
   |
14 |         option if { option = None; false } => { }, //~ ERROR E0302
   |                     ^^^^^^^^^^^^^ assignment in pattern guard
```

--- a/src/librustc_const_eval/check_match.rs
+++ b/src/librustc_const_eval/check_match.rs
@@ -1185,7 +1185,9 @@ impl<'a, 'gcx, 'tcx> Delegate<'tcx> for MutationChecker<'a, 'gcx> {
     fn mutate(&mut self, _: NodeId, span: Span, _: cmt, mode: MutateMode) {
         match mode {
             MutateMode::JustWrite | MutateMode::WriteAndRead => {
-                span_err!(self.cx.tcx.sess, span, E0302, "cannot assign in a pattern guard")
+                struct_span_err!(self.cx.tcx.sess, span, E0302, "cannot assign in a pattern guard")
+                    .span_label(span, &format!("assignment in pattern guard"))
+                    .emit();
             }
             MutateMode::Init => {}
         }

--- a/src/test/compile-fail/E0302.rs
+++ b/src/test/compile-fail/E0302.rs
@@ -12,6 +12,7 @@ fn main() {
     match Some(()) {
         None => { },
         option if { option = None; false } => { }, //~ ERROR E0302
+        //~| NOTE assignment in pattern guard
         Some(_) => { }
     }
 }

